src/convertToObject.js

```diff
@@ -6,7 +6,55 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Methods like `filter` or `map` don't mutate the original array. You can chain these methods directly after the split without assigning them to a new constant first.